main/inc/lib/webservices/Rest.php

```diff
@@ -1590,4 +1595,360 @@ private function encodeParams(array $additionalParams = [])
+            $modelSession['coachId'],
+            $modelSession['sessionCategoryId'],
+            $modelSession['visibility'],
+            false,
+            $modelSession['duration'],
+            $modelSession['description'],
+            $modelSession['showDescription'],
+            $extraFields,
+            $modelSession['sessionAdminId'],
+            $modelSession['sendSubscriptionNotification'],
+            $modelSession['accessUrlId']
+        );
+
+        if (empty($newSessionId)) {
+            throw new Exception(get_lang('SessionNotRegistered'));
+        }
+
+        $promotionId = $modelSession['promotion_id'];
+        if ($promotionId) {
+            $sessionList = array_keys(SessionManager::get_all_sessions_by_promotion($promotionId));
+            $sessionList[] = $newSessionId;
+            SessionManager::subscribe_sessions_to_promotion($modelSession['promotion_id'], $sessionList);
+        }
+
+        $modelExtraFields = [];
+        $fields = SessionManager::getFilteredExtraFields($modelSessionId);
+        if (is_array($fields) and !empty($fields)) {
+            foreach ($fields as $field) {
+                $modelExtraFields[$field['variable']] = $field['value'];
+            }
+        }
+        $allExtraFields = array_merge($modelExtraFields, $extraFields);
+        foreach ($allExtraFields as $name => $value) {
+            if (!SessionManager::update_session_extra_field_value($newSessionId, $name, $value)) {
+                throw new Exception(get_lang('CouldNotUpdateExtraFieldValue'));
+            }
+        }
+
+        $courseList = array_keys(SessionManager::get_course_list_by_session_id($modelSessionId));
+        if (is_array($courseList)
+            && !empty($courseList)
+            && !SessionManager::add_courses_to_session($newSessionId, $courseList)) {
+            throw new Exception(get_lang('CoursesNotAddedToSession'));
+        }
+
+        if (api_is_multiple_url_enabled()) {
+            if (api_get_current_access_url_id() != -1) {
+                UrlManager::add_session_to_url(
+                    $newSessionId,
+                    api_get_current_access_url_id()
+                );
+            } else {
+                UrlManager::add_session_to_url($newSessionId, 1);
+            }
+        } else {
+            UrlManager::add_session_to_url($newSessionId, 1);
+        }
+
+        return [$newSessionId];
+    }
+
+    /**
+     * subscribes a user to a session
+     *
+     * @param int $sessionId the session id
+     * @param string $loginName the user's login name
+     * @return array containing a boolean, whether it worked
+     * @throws Exception
+     */
+    public function subscribeUserToSessionFromUsername($sessionId, $loginName)
+    {
+        if (!SessionManager::isValidId($sessionId)) {
+            throw new Exception(get_lang('SessionNotFound'));
+        }
+
+        $userId = UserManager::get_user_id_from_username($loginName);
+        if (False === $userId) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [TRUE, FALSE and NULL must be lowercase; expected "false" but found "False"](https://app.codacy.com/app/chamilo/chamilo-lms/pullRequest?prid=4554450)